### PR TITLE
feat: pass arbitrary kwargs to docker sdk

### PR DESCRIPTION
**Pull request by JoanFM**
**Changes introduced**
As requested in #1671 , adding arbitrary parameters can be useful to get custom behavior or options for the docker images supporting Peas living in Docker images

### CODE REVIEW

**Comment by tadejsv** [comment]
Oh it was this easy 😂

**Comment by JoanFM** [comment]
> Oh it was this easy

Have to test still, so not sure it was this easy, just a draft

**Comment by tadejsv** [comment]
Would you be able to add a test for using this in a yaml config as well?

**Comment by JoanFM** [comment]
> Would you be able to add a test for using this in a yaml config as well?

I am thinking it won't look very nice, I am trying to see how can I pass direct dictionaries to argparser

**Comment by JoanFM** [comment]
> > > Would you be able to add a test for using this in a yaml config as well?
> > 
> > 
> > I am thinking it won't look very nice, I am trying to see how can I pass direct dictionaries to argparser
> 
> I also came across this problem when thinking how to implement this yesterday. I think the way you did it is probably the best way currently - argparser is unfortunately very limited - you could use something like `eval`, but it's use is discouraged.
> 
> A more fundamental solution would require moving away from Argparser, but that would take a long time

I found a quite good solution with a helper we are already using.

To use it in the yaml files, the syntax would be similar to the usage of `env` in `Flow.yml` as it can be seen in:

`tests/integration/optimizers/flow.yml`

**Comment by tadejsv** [comment]
> > > > Would you be able to add a test for using this in a yaml config as well?
> > > 
> > > 
> > > I am thinking it won't look very nice, I am trying to see how can I pass direct dictionaries to argparser
> > 
> > 
> > I also came across this problem when thinking how to implement this yesterday. I think the way you did it is probably the best way currently - argparser is unfortunately very limited - you could use something like `eval`, but it's use is discouraged.
> > A more fundamental solution would require moving away from Argparser, but that would take a long time
> 
> I found a quite good solution with a helper we are already using.
> 
> To use it in the yaml files, the syntax would be similar to the usage of `env` in `Flow.yml` as it can be seen in:
> 
> `tests/integration/optimizers/flow.yml`

Pretty cool :) I guess for now it is not possible to pass list/dict arguments, right? I mean the following
``` python
docker_kwargs = {'environment': ['VAR1=BAR', 'VAR2=FOO']}

**Comment by JoanFM** [comment]
> > > > > Would you be able to add a test for using this in a yaml config as well?
> > > > 
> > > > 
> > > > I am thinking it won't look very nice, I am trying to see how can I pass direct dictionaries to argparser
> > > 
> > > 
> > > I also came across this problem when thinking how to implement this yesterday. I think the way you did it is probably the best way currently - argparser is unfortunately very limited - you could use something like `eval`, but it's use is discouraged.
> > > A more fundamental solution would require moving away from Argparser, but that would take a long time
> > 
> > 
> > I found a quite good solution with a helper we are already using.
> > To use it in the yaml files, the syntax would be similar to the usage of `env` in `Flow.yml` as it can be seen in:
> > `tests/integration/optimizers/flow.yml`
> 
> Pretty cool :) I guess for now it is not possible to pass list/dict arguments, right? I mean the following [but through command line/yaml]
> 
> ```python
> docker_kwargs = {'environment': ['VAR1=BAR', 'VAR2=FOO']}
> ```

Not sure, should be tested  If u want u can add a tesr and do a PR to this branch or I will try to test tmr

**Comment by tadejsv** [comment]
I can try testing it a bit later yes

**Comment by JoanFM** [comment]
Hey @tadejsv , I did some changes to allow passing strings and arbitrary values, plus I added a test to showcase and document how it can be passed into a `yaml`

**Comment by tadejsv** [comment]
Great work @JoanFM !!

File: daemon/models/custom.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def get_pydantic_fields(config: Union[dict, argparse.ArgumentParser]):
             if hasattr(arg_type, '__self__'):
                 arg_type = type(arg.default) if arg.default else int
             arg_type = str if isinstance(arg_type, argparse.FileType) else arg_type
-            arg_type = dict if type(arg) == KVAppendAction else arg_type
+            arg_type = dict if (isinstance(arg, KVAppendAction) or isinstance(arg, DockerKwargsAppendAction)) else arg_type
```
Comment (tadejsv) [inline daemon/models/custom.py:75]: You can also do `isinstance(arg, (KVAppendAction, DockerKwargsAppendAction))` :)

**Review by rutujasurve94 — APPROVED** [review]

**Pull request merged**